src/main/java/com/google/edith/ShelfDataReader.java

```diff
@@ -0,0 +1,90 @@
+package com.google.edith;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Processes data file of product expiration information to populate future user grocery lists. */
+public class ShelfDataReader {
+
+  /** Finds the specified product in the file. */
+  public String readFile(String itemName) {
+    URL jsonresource = getClass().getClassLoader().getResource("foodkeeper.json");
+    File shelfLifeData = new File(jsonresource.getFile());
+
+    JsonParser jsonParser = new JsonParser();
+    List<JsonArray> potentialMatches = new ArrayList<JsonArray>();
+    try (FileReader reader = new FileReader(shelfLifeData)) {
+      JsonObject data = (JsonObject) jsonParser.parse(reader).getAsJsonObject();
```

Comment:
.parse() is a deprecated method. You can use the static method parseReader() instead.

[](https://javadoc.io/doc/com.google.code.gson/gson/latest/com.google.gson/com/google/gson/JsonParser.html#parseReader(java.io.Reader))